pkg/chrootarchive/go_linux.go

```diff
@@ -0,0 +1,44 @@
+package chrootarchive // import "github.com/docker/docker/pkg/chrootarchive"
+
+import (
+	"runtime"
+
+	"golang.org/x/sys/unix"
+)
+
+// Go starts fn in a goroutine where the root directory, current working
+// directory and umask are unshared from other goroutines and the root directory
+// has been changed to path. These changes are only visible to the goroutine in
+// which fn is executed. Any other goroutines, including ones started from fn,
+// will see the same root directory and file system attributes as the rest of
+// the process.
+func Go(path string, fn func()) error {
```

Comment:
Short blurb following the discussion in the maintainers meeting; suggestion is to not export this if this function has very specific requirements (w.r.t. spawning goroutines - those goroutines would not be chroot'ed)